charts/osbuilder/templates/deployment.yaml

```diff
@@ -98,7 +98,7 @@ kind: Deployment
 metadata:
     labels:
         app.kubernetes.io/name: osbuilder-nginx
```

Comment:
my guess, should we follow the same strategy and use `{{ include "helm-chart.fullname" . }}-nginx` also in the labels? this ensures that multiple instances of osbuilder can be deployed in different namespaces

Reply:
this then winds back down to the cmd args passed to the operator, too.

Reply:
The whole chart should be deployed in a namespace: `helm install -n osbuilder-1 --create-namespace ...`
Then we wouldn't need to worry about name collisions. I looked at that helper and I avoided it because it's supposed to handle the resource name limits. This means, adding a suffix outside the helper may not be safe because we may exceed the limit. It would need another helper that takes the user defined suffix as an argument and ensures the limit is respected. Given the resources are namespaced, I found that unnecessary and skipped it.

Reply:
I tried to deploy a second instance in a different namespace and it fails because of a cluster role name: 

```
~/workspace/kairos/helm-charts (fix-duplicate-deployment-name)$ helm install -n osbuilder2 --create-namespace osbuilder2 charts/osbuilder/
Error: INSTALLATION FAILED: rendered manifests contain a resource that already exists. Unable to continue with install: ClusterRole "osartifactbuilder-operator-manager-role" in namespace "" exists and cannot be imported into the current release: invalid ownership metadata; annotation validation error: key "meta.helm.sh/release-name" must equal "osbuilder2": current value is "osbuilder"; annotation validation error: key "meta.helm.sh/release-namespace" must equal "osbuilder2": current value is "osbuilder"
```

[Others are facing similar issues](https://github.com/kubernetes/ingress-nginx/issues/8936). I wonder if we should make the clusterrole name unique to the helm release too.

Reply:
I commented out the ClusterRole and ClusterRoleBinding resources to see if the deployment name creates any problems and it doesn't seem so:

![image](https://user-images.githubusercontent.com/2794419/210506574-3034a72d-e4d8-4b09-ab9e-8f040386cd63.png)

I suggest we keep this as it is and solve the non-namespaced resource issue on another PR (if at all).